src/backend/macos/utils.rs

```diff
@@ -1,42 +1,42 @@
-use objc::runtime::Object;
-use objc::{class, msg_send, sel, sel_impl};
-
-mod application;
 mod focus_manager;
 mod policy_manager;
-mod url;
-mod window;
-
-pub use application::{INSApplication, NSApplication};
-pub use focus_manager::FocusManager;
-pub use policy_manager::PolicyManager;
-pub use url::{INSURL, NSURL};
-pub use window::{INSWindow, NSWindow};
 
-#[allow(non_upper_case_globals)]
-pub const nil: *mut Object = 0 as *mut _;
+pub use self::focus_manager::FocusManager;
+pub use self::policy_manager::PolicyManager;
 
-pub fn is_main_thread() -> bool {
-    unsafe { msg_send![class!(NSThread), isMainThread] }
-}
+use objc2::rc::Id;
+use objc2_app_kit::{NSApplication, NSView, NSWindow};
+use objc2_foundation::{MainThreadMarker, NSThread};
+use raw_window_handle::RawWindowHandle;
 
 pub fn activate_cocoa_multithreading() {
-    unsafe {
-        let thread: *mut Object = msg_send![class!(NSThread), new];
-        let _: () = msg_send![thread, start];
-    }
+    let thread = NSThread::new();
+    unsafe { thread.start() };
 }
```

Comment:
An example of yet another improvement: This function was previously leaking, as the `NSThread` wasn't released at the end of the scope. `objc2` instead handles this kind of memory management for you automatically.